internal/app/secretless/handlers/mysql/handler.go

```diff
@@ -11,48 +11,127 @@ import (
 	plugin_v1 "github.com/cyberark/secretless-broker/pkg/secretless/plugin/v1"
 )
 
-// BackendConfig stores the connection info to the real backend database.
+// ConnectionDetails stores the connection info to the real backend database.
 // These values are pulled from the handler credentials config
-type BackendConfig struct {
+type ConnectionDetails struct {
 	Host     string
 	Port     uint
 	Username string
 	Password string
 	Options  map[string]string
 }
 
-func (bc *BackendConfig) Address() string {
-	return bc.Host + ":" + strconv.FormatUint(uint64(bc.Port), 10)
+// TODO: all these comments will be moved and subsumed by kumbi's work
+// This format of this packet is described here:
+//
+//   https://dev.mysql.com/doc/internals/en/mysql-packet.html
+//
+//  +-------------+----------------+---------------------------------------------+
+//  |    Type     |      Name      |                 Description                 |
+//	+-------------+----------------+---------------------------------------------+
+//  | int<3>      | payload_length | Length of the payload. The number of bytes  |
+//  |             |                | in the packet beyond the initial 4 bytes    |
+//  |             |                | that make up the packet header.             |
+//  | int<1>      | sequence_id    | Sequence ID                                 |
+//  | string<var> | payload        | [len=payload_length] payload of the packet  |
+//  +-------------+----------------+---------------------------------------------+
+//
+type mysqlPacket []byte
```

Comment:
the package is called mysql - why not just have this be `packet`?

Reply:
Yes that was the plan when we do the merge.